core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -263,7 +265,15 @@ private void drain(Substream substream) {
         }
         if (index == savedState.buffer.size()) { // I'm drained
           state = savedState.substreamDrained(substream);
-          return;
+          onReadyRunnable = new Runnable() {
+            @Override
+            public void run() {
+              if (isReady()) {
+                masterListener.onReady();
```

Comment:
We need to make sure we don't call onReady() if onClosed() is called?

Reply:
Yes, I was about to do it.